storagemarket/deal_handler.go

```diff
@@ -38,7 +41,12 @@ func (d *dealHandler) subscribeUpdates() (event.Subscription, error) {
 	return sub, nil
 }
 
+func (dh *dealHandler) TransferCancelledByUser() bool {
```

Comment:
Should we also have a method that sets transferCancelledByUser to false? eg when the user clicks the "Restart" button in the UI.

Reply:
I think once a transfer is cancelled, best to fail the deal permanently for now and add resumption support here once there is a use case to resume user cancelled transfers.

Reply:
Ok makes sense 👍